packages/parser/test/specs/large-file-memory-leak/large-file-memory-leak.test.ts

```diff
@@ -9,16 +8,20 @@ describe('Large file memory leak protection', { timeout: 20000 }, () => {
     ['cloudflare spec (stringified to a single line)', 'cloudflare-stringified.json'],
     ['cloudflare spec', 'cloudflare.json'],
   ])('%s', async (name, file) => {
-    try {
-      await validate(relativePath(`specs/large-file-memory-leak/${file}`));
+    const res = await validate(relativePath(`specs/large-file-memory-leak/${file}`));
+    if (res.valid === true) {
       assert.fail();
-    } catch (err) {
-      expect(err).toBeInstanceOf(ValidationError);
-      expect(err.message).toMatch(/^OpenAPI schema validation failed.\n(.*)+/);
-      expect(err.message.match(/4xx is not expected to be here!/g) || []).toHaveLength(20);
-      expect(err.message).toContain(
-        'Plus an additional 1016 errors. Please resolve the above and re-run validation to see more.',
-      );
     }
+
+    expect(res.valid).toBe(false);
+    expect(res.errors).toHaveLength(20);
+    expect(res.errors).toStrictEqual(
+      expect.arrayContaining([
+        {
+          message: expect.stringContaining('4xx is not expected to be here!'),
+        },
+      ]),
+    );
+    expect(res.additionalErrors).toBe(1016);
```

Comment:
could we just make all of this into a snapshot test?

```suggestion
    expect(res).toMatchSnapshot();
```